portfolio/src/main/webapp/script.js

```diff
@@ -94,4 +94,27 @@ function changeDropdownVal() {
     const dropdown = document.getElementById("num-comments");
     localStorage["num-comments"] = dropdown.value;
     this.form.submit();
+}
+
+function deleteMessages() {
+    console.log("Fetching response.");
+    const responsePromise = fetch("/delete-data", {method:"POST"});
+    responsePromise.then(handleDelete);
+}
+
+function handleDelete(response) {
+    console.log("Handling the delete response.");
+    const deletePromise = response.text();
+    deletePromise.then(deleteTextFromDom);
+}
+
+function deleteTextFromDom(text) {
+    console.log("Deleting text from DOM: "+text);
+    const table = document.getElementById("comment-section");
+
+    var i;
+    for (i = 0; i < table.children.length; i++) {
+        const comment = table.children[i];
+        comment.remove();
+    }
```

Comment:
Similar to what is said in the training, you can just have the post redirect to the page so it reloads with no data rather than you needing to delete the text from the DOM. But this is probably fine also.

Reply:
I thought deleting was in the other PR and this PR was just for paginating

Reply:
Yeah I messed up making a new branch and those changes ended up in this one. But can you explain what you mean with the redirecting stuff because I don't get it

Reply:
It makes more sense if the post was only trying to delete one or some comments but not all.
If you put a redirect at the end of your post with something like `response.sendRedirect("/...");` then when the page reloads, it will get all the comments that the server knows about and load them onto the page as if you are opening the page for the first time.

Think of the case where multiple people are using your page, though. One person creates a new comment a split second after the other person deleted all comments. Currently the first person would see no comments on the page. If the page redirected to itself afterward, then during the refresh, the  server will notice the new comment and add it to datastore so hopefully by the time. the first user finishes their refresh, they will see the new comment.

Reply:
Ok, I think I understand what you're saying. I think I fixed it, I changed the empty response at the end of the doPost to a redirect back to the page.